plugins/modules/mount.py

```diff
@@ -663,6 +716,41 @@ def get_linux_mounts(module, mntinfo_file="/proc/self/mountinfo"):
     return mounts
 
 
+def _is_same_mount_src(module, args, mntinfo_file="/proc/self/mountinfo"):
+    """Return True if the mounted fs on mountpoint is the same source than src"""
+    mountpoint = args['name']
+    src = args['src']
+    lines = _get_mount_info(module)
```

Comment:
See [comment](https://github.com/ansible-collections/ansible.posix/pull/267/files/13916108fa32a53804c2e2698afe33565704b939#r830526629): this call (or the `_get_mount_info` function) must be system aware